example/weasel/grammar.ebnf

```diff
@@ -0,0 +1,154 @@
+(*
+    In the grammar below, the following notations are adopted:
+        - The nonterminal symbol `u32` denotes 32-bit unsigned integers.
+        - The nonterminal symbol `integer` denotes arbitrary-precision mathematical integers.
+        - The nonterminal symbol `real` denotes arbitrary-precision mathematical real numbers.
+*)
+
+idchar =
+    "0" | ... | "9"
+  | "A" | ... | "Z"
+  | "a" | ... | "z"
+  | "!" | "#" | "$" | "%" | "&" | "'" | "*" | "+" | "-" | "." | "/"
+  | ":" | "<" | "=" | ">" | "?" | "@" | "\" | "^" | "_" | "`" | "|" | "~" ;
+
+id =
+  idchar , { idchar } ;
+
+ind =
+    "$" , id
+  | u32 ;
+
+unop =
+    "clz"             (on wasm_int_type)
+  | "ctz"             (on wasm_int_type)
+  | "popcnt"          (on wasm_int_type)
+  | "abs"
+  | "neg"
+  | "sqrt"
+  | "ceil"
+  | "floor"
+  | "trunc"           (on wasm_float_type or real)
+  | "nearst" ;        (on wasm_float_type or real)
+
+binop =
+    "add" | "+"
+  | "sub" | "-"
+  | "mul" | "*"
+  | "div" | "/"
+  | "rem" | "%"       (on wasm_int_type or integer)
+  | "min"
+  | "max"
+  | "and" | "&"       (on wasm_int_type or integer)
+  | "or" | "|"        (on wasm_int_type or integer)
+  | "xor" | "^"       (on wasm_int_type or integer)
+  | "shl" | "<<"      (on wasm_int_type or integer)
+  | "shr_s" | ">>"    (on wasm_int_type or integer)
+  | "shr_x" | ">>>"   (on wasm_int_type or integer)
```

Comment:
Many of these are quite unusual on mathematical integers, are there similar operators in WhyML/Gospel/ACSL?

Reply:
Fine with me.